core/src/main/java/feast/core/model/FeatureStatistics.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.io.*;
+import java.util.Date;
+import java.util.List;
+import javax.persistence.*;
+import javax.persistence.Entity;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.tensorflow.metadata.v0.*;
+
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(
+    name = "feature_statistics",
+    indexes = {
+      @Index(
+          name = "idx_feature_statistics_feature",
+          columnList = "project,feature_set,version,name"),
+      @Index(name = "idx_feature_statistics_dataset_id", columnList = "datasetId"),
+      @Index(name = "idx_feature_statistics_date", columnList = "date"),
+    })
+public class FeatureStatistics {
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private int id;
+
+  @ManyToOne
+  @JoinColumns({
+    @JoinColumn(name = "project", referencedColumnName = "project"),
+    @JoinColumn(name = "feature_set", referencedColumnName = "feature_set"),
+    @JoinColumn(name = "version", referencedColumnName = "version"),
+    @JoinColumn(name = "name", referencedColumnName = "name")
+  })
+  private Feature feature;
+
+  // Only one of these fields should be populated.
```

Comment:
Previously when we discussed this we said we would store statistics with a key of `day` + `dataset`, which would allow us to roll it up to any time range and to any dataset.

Is there a reason why both won't be set here? Are we still doing that?

Reply:
a large portion of statistics are lost in aggregation, so we want to avoid aggregating within datasets, which should be the logical unit for retrieval here anyway.

Reply:
Have we documented the list of statistics that are lost? How are we dealing with that in the case of a time range query?